chain/substreams/src/trigger.rs

```diff
@@ -189,95 +183,37 @@ where
         _subgraph_metrics: &Arc<graph::prelude::SubgraphInstanceMetrics>,
         _instrument: bool,
     ) -> Result<BlockState<Chain>, MappingError> {
-        for entity_change in block.changes.entity_changes.iter() {
-            match entity_change.operation() {
-                Operation::Unset => {
+        for parsed_change in block.parsed_changes.clone().into_iter() {
```

Comment:
This `clone` seems at first sight superflous. In each arm of the match, you either use a reference or clone, I didn't see any move in there.

Reply:
I will check but I think this is because block is a &Arc so you can't move it out of there to call into_iter

Reply:
Seems that `state.entity_cache.remove(entity_key);` is the only move. I think cloning just this one would be better. I leave to you the decision.

Reply:
Scratch all my comment above, I just saw `state.entity_cache.set(key, entity)?;` which is a full move, I missed it because it was part of non-modified code. Sorry :)

Reply:
Yeah that's it I was just re-checking and the set requires the move so we might as well clone it all ahead of time. No problem at all, thanks for checking!